<a href="https://colab.research.google.com/github/UdaraChamidu/Parkinson-s-Disease-Classification-Model/blob/main/2021E049_Lab01.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# **Parkinson's Disease classification model**




In [30]:
# import useful libraries
import numpy as np
import pandas as pd

In [31]:
# load csv file into variable "data"
data = pd.read_csv('data.csv',header=None)

In [32]:
# display data
data

0            1             2             3               4   \
0              name  MDVP:Fo(Hz)  MDVP:Fhi(Hz)  MDVP:Flo(Hz)  MDVP:Jitter(%)   
1    phon_R01_S01_1      119.992       157.302        74.997         0.00784   
2    phon_R01_S01_2        122.4        148.65       113.819         0.00968   
3    phon_R01_S01_3      116.682       131.111       111.555          0.0105   
4    phon_R01_S01_4      116.676       137.871       111.366         0.00997   
..              ...          ...           ...           ...             ...   
191  phon_R01_S50_2      174.188       230.978        94.261         0.00459   
192  phon_R01_S50_3      209.516       253.017        89.488         0.00564   
193  phon_R01_S50_4      174.688       240.005        74.287          0.0136   
194  phon_R01_S50_5      198.764       396.961        74.904          0.0074   
195  phon_R01_S50_6      214.289       260.277        77.973         0.00567   

                   5         6         7           8             9   ...  \
0    MDVP:Jitter(Abs)  MDVP:RAP  MDVP:PPQ  Jitter:DDP  MDVP:Shimmer  ...   
1             0.00007    0.0037   0.00554     0.01109       0.04374  ...   
2             0.00008   0.00465   0.00696     0.01394       0.06134  ...   
3             0.00009   0.00544   0.00781     0.01633       0.05233  ...   
4             0.00009   0.00502   0.00698     0.01505       0.05492  ...   
..                ...       ...       ...         ...           ...  ...   
191           0.00003   0.00263   0.00259      0.0079       0.04087  ...   
192           0.00003   0.00331   0.00292     0.00994       0.02751  ...   
193           0.00008   0.00624   0.00564     0.01873       0.02308  ...   
194           0.00004    0.0037    0.0039     0.01109       0.02296  ...   
195           0.00003   0.00295   0.00317     0.00885       0.01884  ...   

              14       15      16      17        18        19         20  \
0    Shimmer:DDA      NHR     HNR  status      RPDE       DFA    spread1   
1        0.06545  0.02211  21.033       1  0.414783  0.815285  -4.813031   
2        0.09403  0.01929  19.085       1  0.458359  0.819521  -4.075192   
3         0.0827  0.01309  20.651       1  0.429895  0.825288  -4.443179   
4        0.08771  0.01353  20.644       1  0.434969  0.819235  -4.117501   
..           ...      ...     ...     ...       ...       ...        ...   
191      0.07008  0.02764  19.517       0  0.448439  0.657899  -6.538586   
192      0.04812   0.0181  19.147       0  0.431674  0.683244  -6.195325   
193      0.03804  0.10715  17.883       0  0.407567  0.655683  -6.787197   
194      0.03794  0.07223   19.02       0  0.451221  0.643956  -6.744577   
195      0.03078  0.04398  21.209       0  0.462803  0.664357  -5.724056   

           21        22        23  
0     spread2        D2       PPE  
1    0.266482  2.301442  0.284654  
2     0.33559  2.486855  0.368674  
3    0.311173  2.342259  0.332634  
4    0.334147  2.405554  0.368975  
..        ...       ...       ...  
191  0.121952  2.657476   0.13305  
192  0.129303  2.784312  0.168895  
193  0.158453  2.679772  0.131728  
194  0.207454  2.138608  0.123306  
195  0.190667  2.555477  0.148569  

[196 rows x 24 columns]

In [33]:
# feature names
all_features = data.iloc[0]
excluded_columns = [0, 17]
selected_features = [all_features[i] for i in range(len(all_features)) if i not in excluded_columns]
print("Features : ", selected_features)

Features :  ['MDVP:Fo(Hz)', 'MDVP:Fhi(Hz)', 'MDVP:Flo(Hz)', 'MDVP:Jitter(%)', 'MDVP:Jitter(Abs)', 'MDVP:RAP', 'MDVP:PPQ', 'Jitter:DDP', 'MDVP:Shimmer', 'MDVP:Shimmer(dB)', 'Shimmer:APQ3', 'Shimmer:APQ5', 'MDVP:APQ', 'Shimmer:DDA', 'NHR', 'HNR', 'RPDE', 'DFA', 'spread1', 'spread2', 'D2', 'PPE']


In [34]:
# disply data
data

0            1             2             3               4   \
0              name  MDVP:Fo(Hz)  MDVP:Fhi(Hz)  MDVP:Flo(Hz)  MDVP:Jitter(%)   
1    phon_R01_S01_1      119.992       157.302        74.997         0.00784   
2    phon_R01_S01_2        122.4        148.65       113.819         0.00968   
3    phon_R01_S01_3      116.682       131.111       111.555          0.0105   
4    phon_R01_S01_4      116.676       137.871       111.366         0.00997   
..              ...          ...           ...           ...             ...   
191  phon_R01_S50_2      174.188       230.978        94.261         0.00459   
192  phon_R01_S50_3      209.516       253.017        89.488         0.00564   
193  phon_R01_S50_4      174.688       240.005        74.287          0.0136   
194  phon_R01_S50_5      198.764       396.961        74.904          0.0074   
195  phon_R01_S50_6      214.289       260.277        77.973         0.00567   

                   5         6         7           8             9   ...  \
0    MDVP:Jitter(Abs)  MDVP:RAP  MDVP:PPQ  Jitter:DDP  MDVP:Shimmer  ...   
1             0.00007    0.0037   0.00554     0.01109       0.04374  ...   
2             0.00008   0.00465   0.00696     0.01394       0.06134  ...   
3             0.00009   0.00544   0.00781     0.01633       0.05233  ...   
4             0.00009   0.00502   0.00698     0.01505       0.05492  ...   
..                ...       ...       ...         ...           ...  ...   
191           0.00003   0.00263   0.00259      0.0079       0.04087  ...   
192           0.00003   0.00331   0.00292     0.00994       0.02751  ...   
193           0.00008   0.00624   0.00564     0.01873       0.02308  ...   
194           0.00004    0.0037    0.0039     0.01109       0.02296  ...   
195           0.00003   0.00295   0.00317     0.00885       0.01884  ...   

              14       15      16      17        18        19         20  \
0    Shimmer:DDA      NHR     HNR  status      RPDE       DFA    spread1   
1        0.06545  0.02211  21.033       1  0.414783  0.815285  -4.813031   
2        0.09403  0.01929  19.085       1  0.458359  0.819521  -4.075192   
3         0.0827  0.01309  20.651       1  0.429895  0.825288  -4.443179   
4        0.08771  0.01353  20.644       1  0.434969  0.819235  -4.117501   
..           ...      ...     ...     ...       ...       ...        ...   
191      0.07008  0.02764  19.517       0  0.448439  0.657899  -6.538586   
192      0.04812   0.0181  19.147       0  0.431674  0.683244  -6.195325   
193      0.03804  0.10715  17.883       0  0.407567  0.655683  -6.787197   
194      0.03794  0.07223   19.02       0  0.451221  0.643956  -6.744577   
195      0.03078  0.04398  21.209       0  0.462803  0.664357  -5.724056   

           21        22        23  
0     spread2        D2       PPE  
1    0.266482  2.301442  0.284654  
2     0.33559  2.486855  0.368674  
3    0.311173  2.342259  0.332634  
4    0.334147  2.405554  0.368975  
..        ...       ...       ...  
191  0.121952  2.657476   0.13305  
192  0.129303  2.784312  0.168895  
193  0.158453  2.679772  0.131728  
194  0.207454  2.138608  0.123306  
195  0.190667  2.555477  0.148569  

[196 rows x 24 columns]

In [35]:
# change data into target and features
x = data.iloc[1:, 1:-1]
y = data.iloc[1:, 17]

In [36]:
# view features
x

1        2        3        4        5        6        7        8   \
1    119.992  157.302   74.997  0.00784  0.00007   0.0037  0.00554  0.01109   
2      122.4   148.65  113.819  0.00968  0.00008  0.00465  0.00696  0.01394   
3    116.682  131.111  111.555   0.0105  0.00009  0.00544  0.00781  0.01633   
4    116.676  137.871  111.366  0.00997  0.00009  0.00502  0.00698  0.01505   
5    116.014  141.781  110.655  0.01284  0.00011  0.00655  0.00908  0.01966   
..       ...      ...      ...      ...      ...      ...      ...      ...   
191  174.188  230.978   94.261  0.00459  0.00003  0.00263  0.00259   0.0079   
192  209.516  253.017   89.488  0.00564  0.00003  0.00331  0.00292  0.00994   
193  174.688  240.005   74.287   0.0136  0.00008  0.00624  0.00564  0.01873   
194  198.764  396.961   74.904   0.0074  0.00004   0.0037   0.0039  0.01109   
195  214.289  260.277   77.973  0.00567  0.00003  0.00295  0.00317  0.00885   

          9      10  ...       13       14       15      16 17        18  \
1    0.04374  0.426  ...  0.02971  0.06545  0.02211  21.033  1  0.414783   
2    0.06134  0.626  ...  0.04368  0.09403  0.01929  19.085  1  0.458359   
3    0.05233  0.482  ...   0.0359   0.0827  0.01309  20.651  1  0.429895   
4    0.05492  0.517  ...  0.03772  0.08771  0.01353  20.644  1  0.434969   
5    0.06425  0.584  ...  0.04465   0.1047  0.01767  19.649  1  0.417356   
..       ...    ...  ...      ...      ...      ...     ... ..       ...   
191  0.04087  0.405  ...  0.02745  0.07008  0.02764  19.517  0  0.448439   
192  0.02751  0.263  ...  0.01879  0.04812   0.0181  19.147  0  0.431674   
193  0.02308  0.256  ...  0.01667  0.03804  0.10715  17.883  0  0.407567   
194  0.02296  0.241  ...  0.01588  0.03794  0.07223   19.02  0  0.451221   
195  0.01884   0.19  ...  0.01373  0.03078  0.04398  21.209  0  0.462803   

           19         20        21        22  
1    0.815285  -4.813031  0.266482  2.301442  
2    0.819521  -4.075192   0.33559  2.486855  
3    0.825288  -4.443179  0.311173  2.342259  
4    0.819235  -4.117501  0.334147  2.405554  
5    0.823484  -3.747787  0.234513   2.33218  
..        ...        ...       ...       ...  
191  0.657899  -6.538586  0.121952  2.657476  
192  0.683244  -6.195325  0.129303  2.784312  
193  0.655683  -6.787197  0.158453  2.679772  
194  0.643956  -6.744577  0.207454  2.138608  
195  0.664357  -5.724056  0.190667  2.555477  

[195 rows x 22 columns]

In [37]:
# view target column
y

1      1
2      1
3      1
4      1
5      1
      ..
191    0
192    0
193    0
194    0
195    0
Name: 17, Length: 195, dtype: object

In [40]:
# check for duplicate values
data.duplicated()

0      False
1      False
2      False
3      False
4      False
       ...  
191    False
192    False
193    False
194    False
195    False
Length: 196, dtype: bool

In [39]:
# check for null data
data.isnull().sum()

0     0
1     0
2     0
3     0
4     0
5     0
6     0
7     0
8     0
9     0
10    0
11    0
12    0
13    0
14    0
15    0
16    0
17    0
18    0
19    0
20    0
21    0
22    0
23    0
dtype: int64

In [42]:
# access target column
data['target'] = data.iloc[:, 17]

In [43]:
data.head()

0            1             2             3               4  \
0            name  MDVP:Fo(Hz)  MDVP:Fhi(Hz)  MDVP:Flo(Hz)  MDVP:Jitter(%)   
1  phon_R01_S01_1      119.992       157.302        74.997         0.00784   
2  phon_R01_S01_2        122.4        148.65       113.819         0.00968   
3  phon_R01_S01_3      116.682       131.111       111.555          0.0105   
4  phon_R01_S01_4      116.676       137.871       111.366         0.00997   

                  5         6         7           8             9  ...  \
0  MDVP:Jitter(Abs)  MDVP:RAP  MDVP:PPQ  Jitter:DDP  MDVP:Shimmer  ...   
1           0.00007    0.0037   0.00554     0.01109       0.04374  ...   
2           0.00008   0.00465   0.00696     0.01394       0.06134  ...   
3           0.00009   0.00544   0.00781     0.01633       0.05233  ...   
4           0.00009   0.00502   0.00698     0.01505       0.05492  ...   

        15      16      17        18        19         20        21        22  \
0      NHR     HNR  status      RPDE       DFA    spread1   spread2        D2   
1  0.02211  21.033       1  0.414783  0.815285  -4.813031  0.266482  2.301442   
2  0.01929  19.085       1  0.458359  0.819521  -4.075192   0.33559  2.486855   
3  0.01309  20.651       1  0.429895  0.825288  -4.443179  0.311173  2.342259   
4  0.01353  20.644       1  0.434969  0.819235  -4.117501  0.334147  2.405554   

         23  target  
0       PPE  status  
1  0.284654       1  
2  0.368674       1  
3  0.332634       1  
4  0.368975       1  

[5 rows x 25 columns]

In [44]:
# Drop column
data = data.drop(data.columns[17], axis=1)

In [45]:
data.head()

0            1             2             3               4  \
0            name  MDVP:Fo(Hz)  MDVP:Fhi(Hz)  MDVP:Flo(Hz)  MDVP:Jitter(%)   
1  phon_R01_S01_1      119.992       157.302        74.997         0.00784   
2  phon_R01_S01_2        122.4        148.65       113.819         0.00968   
3  phon_R01_S01_3      116.682       131.111       111.555          0.0105   
4  phon_R01_S01_4      116.676       137.871       111.366         0.00997   

                  5         6         7           8             9  ...  \
0  MDVP:Jitter(Abs)  MDVP:RAP  MDVP:PPQ  Jitter:DDP  MDVP:Shimmer  ...   
1           0.00007    0.0037   0.00554     0.01109       0.04374  ...   
2           0.00008   0.00465   0.00696     0.01394       0.06134  ...   
3           0.00009   0.00544   0.00781     0.01633       0.05233  ...   
4           0.00009   0.00502   0.00698     0.01505       0.05492  ...   

            14       15      16        18        19         20        21  \
0  Shimmer:DDA      NHR     HNR      RPDE       DFA    spread1   spread2   
1      0.06545  0.02211  21.033  0.414783  0.815285  -4.813031  0.266482   
2      0.09403  0.01929  19.085  0.458359  0.819521  -4.075192   0.33559   
3       0.0827  0.01309  20.651  0.429895  0.825288  -4.443179  0.311173   
4      0.08771  0.01353  20.644  0.434969  0.819235  -4.117501  0.334147   

         22        23  target  
0        D2       PPE  status  
1  2.301442  0.284654       1  
2  2.486855  0.368674       1  
3  2.342259  0.332634       1  
4  2.405554  0.368975       1  

[5 rows x 24 columns]

In [46]:
# save modified data set
data.to_csv('modified_data.csv',index=False)


In [47]:
data = data.drop(0)
# unnessory row dropped

In [48]:
# find number of unique values in 'target' column
unique_values = data['target'].nunique()
print(f"Number of unique values in the 'target' column : {unique_values}")

Number of unique values in the 'target' column : 2


In [49]:
# devide into training and testing data sets
from sklearn.model_selection import train_test_split
x_train, x_test, y_train, y_test = train_test_split(x, y, test_size=0.2, random_state=42)

In [52]:
# check data overlaping or not
overlap = set(x_train.index) & set(x_test.index)
print("Overlap between train and test sets:", len(overlap))

Overlap between train and test sets: 0


In [53]:
# view training x data
x_train

1        2        3        4        5        6        7        8   \
6    120.552  131.162  113.787  0.00968  0.00008  0.00463   0.0075  0.01388   
136  110.453  127.611  105.554  0.00494  0.00004  0.00244  0.00315  0.00731   
123   138.19  203.522    83.34  0.00704  0.00005  0.00406  0.00398  0.01218   
168  260.105  264.919  237.303  0.00339  0.00001  0.00205  0.00186  0.00616   
86   180.978  200.125  155.495  0.00406  0.00002   0.0022  0.00244  0.00659   
..       ...      ...      ...      ...      ...      ...      ...      ...   
107  155.078  163.736  144.148  0.00168  0.00001  0.00068  0.00092  0.00204   
15   152.845  163.305   75.836  0.00294  0.00002  0.00121  0.00149  0.00364   
93   148.272  164.989  142.299  0.00459  0.00003   0.0025  0.00256   0.0075   
180  148.143  155.982  135.041  0.00392  0.00003  0.00204  0.00231  0.00612   
103  139.224  586.567   66.157  0.03011  0.00022  0.01854  0.01628  0.05563   

          9      10  ...       13       14       15      16 17        18  \
6    0.04701  0.456  ...  0.03243  0.06985  0.01222  21.378  1  0.415564   
136  0.04128  0.379  ...  0.03316  0.06688  0.02529  17.707  1  0.653427   
123  0.04479  0.441  ...   0.0322  0.07761  0.01968  18.305  1  0.538016   
168   0.0203  0.197  ...  0.01367  0.03557   0.0091  21.083  0  0.440988   
86   0.03852  0.331  ...  0.02877  0.06321  0.02782  16.176  1  0.583574   
..       ...    ...  ...      ...      ...      ...     ... ..       ...   
107  0.01064  0.097  ...  0.00928  0.01567  0.00233  29.746  1  0.334171   
15   0.01828  0.158  ...  0.01246  0.03191  0.00609  24.922  1  0.474791   
93    0.0419  0.383  ...  0.03051   0.0715  0.01914   18.78  1  0.454444   
180   0.0145  0.131  ...  0.01263  0.02175   0.0054  23.683  1  0.398499   
103  0.09419   0.93  ...  0.06023  0.16654   0.2593  10.489  1  0.596362   

           19         20        21        22  
6    0.825069  -4.242867  0.299111   2.18756  
136  0.706687  -5.333619  0.322044  2.631793  
123   0.74148  -5.418787  0.160267  2.090438  
168  0.628058  -7.517934  0.160414  1.881767  
86   0.727747  -5.657899  0.315903  3.098256  
..        ...        ...       ...       ...  
107   0.67793  -6.981201   0.18455  2.129924  
15   0.654027  -6.105098  0.203653  2.125618  
93   0.734504  -5.952058   0.08784  2.344336  
180  0.778349  -5.711205  0.240875  2.845109  
103  0.641418  -3.269487  0.270641  2.690917  

[156 rows x 22 columns]

In [54]:
# view training y data
y_train

6      1
136    1
123    1
168    0
86     1
      ..
107    1
15     1
93     1
180    1
103    1
Name: 17, Length: 156, dtype: object

In [55]:
# view testing x data
x_test

1        2        3        4         5        6        7        8   \
139  112.239  126.609  104.095  0.00472   0.00004  0.00238   0.0029  0.00715   
17   144.188  349.259   82.764  0.00544   0.00004  0.00211  0.00292  0.00632   
156   117.87  127.349   95.654  0.00647   0.00005  0.00356    0.003  0.01067   
97   159.116  168.913  144.811  0.00342   0.00002  0.00178  0.00184  0.00535   
69   143.533  162.215   65.809  0.01101   0.00008  0.00647  0.00467  0.01941   
154  121.345  139.644    98.25  0.00684   0.00006  0.00388  0.00332  0.01164   
56    109.86  126.358  104.437  0.00874   0.00008  0.00398  0.00539  0.01193   
16   142.167  217.455   83.159  0.00369   0.00003  0.00157  0.00203  0.00471   
113  204.664    221.3  189.621  0.00841   0.00004  0.00502  0.00485  0.01506   
112  208.519  220.315   199.02  0.00609   0.00003  0.00368  0.00339  0.01105   
185  116.848  217.552   99.503  0.00531   0.00005   0.0026  0.00346   0.0078   
19   153.046  175.829   68.623  0.00742   0.00005  0.00364  0.00432  0.01092   
83    100.96  110.019   95.628  0.00606   0.00006  0.00351  0.00348  0.01053   
10    95.056  120.103   91.226  0.00532   0.00006  0.00268  0.00332  0.00803   
165  102.273   142.83   85.902  0.00907   0.00009  0.00493  0.00461   0.0148   
118  170.756  450.247   79.032  0.00555   0.00003  0.00244  0.00261  0.00731   
70    148.09  162.824   67.343  0.00762   0.00005  0.00467  0.00354    0.014   
114  210.141  232.706  185.258  0.00534   0.00003  0.00321   0.0028  0.00964   
193  174.688  240.005   74.287   0.0136   0.00008  0.00624  0.00564  0.01873   
120  217.116  233.481   93.978  0.00404   0.00002  0.00127  0.00128  0.00381   
124  182.018  197.173   79.187  0.00842   0.00005  0.00506  0.00449  0.01517   
145  202.544   241.35  164.168  0.00254   0.00001    0.001  0.00133  0.00301   
67   140.341  159.774   67.021  0.00817   0.00006   0.0043   0.0044  0.01289   
46   242.852  255.034  227.911  0.00225  0.000009  0.00117  0.00139   0.0035   
159  126.144  154.284   97.543  0.00975   0.00008  0.00593  0.00454  0.01778   
116  151.872  492.892   69.085  0.00856   0.00006  0.00404  0.00385  0.01211   
68   136.969  166.607   66.004  0.00923   0.00007  0.00507  0.00463   0.0152   
94   152.125  161.469   76.596  0.00382   0.00003  0.00191  0.00226  0.00574   
31   197.076  206.896  192.055  0.00289   0.00001  0.00166  0.00168  0.00498   
102  128.451  150.449   75.632  0.01551   0.00012  0.00905  0.00909  0.02716   
119  178.285  442.824   82.063  0.00462   0.00003  0.00157  0.00194  0.00472   
76   110.707  122.611  105.007  0.00516   0.00005  0.00277  0.00289  0.00831   
25   163.656  200.841   76.779  0.00742   0.00005   0.0038   0.0039   0.0114   
173  110.739  113.597  100.139  0.00356   0.00003   0.0017    0.002   0.0051   
128  166.888  198.966   79.512  0.00638   0.00004  0.00368  0.00351  0.01104   
170  240.301  245.135  219.783  0.00517   0.00002  0.00316  0.00283  0.00949   
20   156.405  189.398  142.822  0.00768   0.00005  0.00372  0.00399  0.01116   
169  197.569  217.627   90.794  0.00803   0.00004   0.0049  0.00448   0.0147   
74   112.014  588.518  107.024  0.00533   0.00005  0.00268  0.00329  0.00805   

          9      10  ...       13       14       15      16 17        18  \
139  0.05643  0.517  ...  0.04451  0.09211  0.02629  17.366  1  0.640945   
17   0.02047  0.192  ...  0.02074  0.02908  0.01859  22.333  1   0.56738   
156  0.03087  0.276  ...  0.02402  0.04977  0.02631  22.431  1    0.6283   
97   0.03381  0.307  ...  0.02809  0.05417  0.00852  22.663  1  0.366329   
69   0.05384  0.478  ...  0.03392  0.09455  0.04882  20.338  1  0.513237   
154  0.02534  0.241  ...  0.02056  0.04019  0.04179   21.52  1  0.566867   
56   0.03209  0.307  ...  0.02454  0.05368   0.0118  20.767  1  0.558586   
16   0.01503  0.126  ...  0.01359  0.02316  0.00839  25.175  1  0.565924   
113  0.02378   0.21  ...  0.01767  0.03831  0.01316  21.305  1  0.498877   
112  0.01761  0.155  ...  0.01307  0.02855   0.00

In [56]:
# view testing t data
y_test

139    1
17     1
156    1
97     1
69     1
154    1
56     1
16     1
113    1
112    1
185    0
19     1
83     1
10     1
165    1
118    1
70     1
114    1
193    0
120    1
124    1
145    1
67     1
46     0
159    1
116    1
68     1
94     1
31     0
102    1
119    1
76     1
25     1
173    0
128    1
170    0
20     1
169    0
74     1
Name: 17, dtype: object

In [57]:
# import essential things for algorithm and performance calculation
from sklearn.model_selection import train_test_split
from sklearn.neighbors import KNeighborsClassifier
from sklearn.metrics import accuracy_score,classification_report,confusion_matrix

# ***K-Nearest Neighbors (KNN) algorithm***

In [ ]:
knn = KNeighborsClassifier(n_neighbors=5)
knn.fit(x_train, y_train)
y_pred = knn.predict(x_test)

# Accuracy, Confusion_matrix, Classification_report
accuracy = accuracy_score(y_test, y_pred)
conf_matrix = confusion_matrix(y_test, y_pred)
class_report = classification_report(y_test, y_pred)

print("PERFORMANCE OF THE ALGORITHM")
print(f"\nAccuracy : {accuracy}")
print(f"\nConfusion Matrix :\n\n{conf_matrix}")
print(f"\nClassification Report :\n\n{class_report}")



PERFORMANCE OF THE ALGORITHM

Accuracy : 0.8205128205128205

Confusion Matrix :

[[ 3  4]
 [ 3 29]]

Classification Report :

              precision    recall  f1-score   support

           0       0.50      0.43      0.46         7
           1       0.88      0.91      0.89        32

    accuracy                           0.82        39
   macro avg       0.69      0.67      0.68        39
weighted avg       0.81      0.82      0.81        39



# **Decision Tree Algorithm**

In [ ]:
from sklearn.tree import DecisionTreeClassifier

dtree = DecisionTreeClassifier(random_state=42)
dtree.fit(x_train, y_train)
y_pred = dtree.predict(x_test)

# Accuracy, Confusion_matrix, Classification_report
accuracy = accuracy_score(y_test, y_pred)
conf_matrix = confusion_matrix(y_test, y_pred)
class_report = classification_report(y_test, y_pred)

print("PERFORMANCE OF THE ALGORITHM")
print(f"\nAccuracy : {accuracy}")
print(f"\nConfusion Matrix :\n\n{conf_matrix}")
print(f"\nClassification Report :\n\n{class_report}")

PERFORMANCE OF THE ALGORITHM

Accuracy : 1.0

Confusion Matrix :

[[ 7  0]
 [ 0 32]]

Classification Report :

              precision    recall  f1-score   support

           0       1.00      1.00      1.00         7
           1       1.00      1.00      1.00        32

    accuracy                           1.00        39
   macro avg       1.00      1.00      1.00        39
weighted avg       1.00      1.00      1.00        39



# **Support Vector Machine Algorithm**

In [ ]:
from sklearn.svm import SVC

svm = SVC(random_state= 42)
svm.fit(x_train, y_train)
y_pred = dtree.predict(x_test)

# Accuracy, Confusion_matrix, Classification_report
accuracy = accuracy_score(y_test, y_pred)
conf_matrix = confusion_matrix(y_test, y_pred)
class_report = classification_report(y_test, y_pred)

print("PERFORMANCE OF THE ALGORITHM")
print(f"\nAccuracy : {accuracy}")
print(f"\nConfusion Matrix :\n\n{conf_matrix}")
print(f"\nClassification Report :\n\n{class_report}")

PERFORMANCE OF THE ALGORITHM

Accuracy : 1.0

Confusion Matrix :

[[ 7  0]
 [ 0 32]]

Classification Report :

              precision    recall  f1-score   support

           0       1.00      1.00      1.00         7
           1       1.00      1.00      1.00        32

    accuracy                           1.00        39
   macro avg       1.00      1.00      1.00        39
weighted avg       1.00      1.00      1.00        39



In [ ]:
data

0        1        2        3        4        5        6  \
1    phon_R01_S01_1  119.992  157.302   74.997  0.00784  0.00007   0.0037   
2    phon_R01_S01_2    122.4   148.65  113.819  0.00968  0.00008  0.00465   
3    phon_R01_S01_3  116.682  131.111  111.555   0.0105  0.00009  0.00544   
4    phon_R01_S01_4  116.676  137.871  111.366  0.00997  0.00009  0.00502   
5    phon_R01_S01_5  116.014  141.781  110.655  0.01284  0.00011  0.00655   
..              ...      ...      ...      ...      ...      ...      ...   
191  phon_R01_S50_2  174.188  230.978   94.261  0.00459  0.00003  0.00263   
192  phon_R01_S50_3  209.516  253.017   89.488  0.00564  0.00003  0.00331   
193  phon_R01_S50_4  174.688  240.005   74.287   0.0136  0.00008  0.00624   
194  phon_R01_S50_5  198.764  396.961   74.904   0.0074  0.00004   0.0037   
195  phon_R01_S50_6  214.289  260.277   77.973  0.00567  0.00003  0.00295   

           7        8        9  ...       14       15      16        18  \
1    0.00554  0.01109  0.04374  ...  0.06545  0.02211  21.033  0.414783   
2    0.00696  0.01394  0.06134  ...  0.09403  0.01929  19.085  0.458359   
3    0.00781  0.01633  0.05233  ...   0.0827  0.01309  20.651  0.429895   
4    0.00698  0.01505  0.05492  ...  0.08771  0.01353  20.644  0.434969   
5    0.00908  0.01966  0.06425  ...   0.1047  0.01767  19.649  0.417356   
..       ...      ...      ...  ...      ...      ...     ...       ...   
191  0.00259   0.0079  0.04087  ...  0.07008  0.02764  19.517  0.448439   
192  0.00292  0.00994  0.02751  ...  0.04812   0.0181  19.147  0.431674   
193  0.00564  0.01873  0.02308  ...  0.03804  0.10715  17.883  0.407567   
194   0.0039  0.01109  0.02296  ...  0.03794  0.07223   19.02  0.451221   
195  0.00317  0.00885  0.01884  ...  0.03078  0.04398  21.209  0.462803   

           19         20        21        22        23 target  
1    0.815285  -4.813031  0.266482  2.301442  0.284654      1  
2    0.819521  -4.075192   0.33559  2.486855  0.368674      1  
3    0.825288  -4.443179  0.311173  2.342259  0.332634      1  
4    0.819235  -4.117501  0.334147  2.405554  0.368975      1  
5    0.823484  -3.747787  0.234513   2.33218  0.410335      1  
..        ...        ...       ...       ...       ...    ...  
191  0.657899  -6.538586  0.121952  2.657476   0.13305      0  
192  0.683244  -6.195325  0.129303  2.784312  0.168895      0  
193  0.655683  -6.787197  0.158453  2.679772  0.131728      0  
194  0.643956  -6.744577  0.207454  2.138608  0.123306      0  
195  0.664357  -5.724056  0.190667  2.555477  0.148569      0  

[195 rows x 24 columns]

**Answers**

4. Describe the selected dataset in detail





*   Features :



 Features are predictors or independent variables in the dataset. These are the names of the data that represent all the columns except the target column.

 There are 22 columns in my dataset except target column. Those are

 ['MDVP:Fo(Hz)', 'MDVP:Fhi(Hz)', 'MDVP:Flo(Hz)', 'MDVP:Jitter(%)', 'MDVP:Jitter(Abs)', 'MDVP:RAP', 'MDVP:PPQ', 'Jitter:DDP', 'MDVP:Shimmer', 'MDVP:Shimmer(dB)', 'Shimmer:APQ3', 'Shimmer:APQ5', 'MDVP:APQ', 'Shimmer:DDA', 'NHR', 'HNR', 'RPDE', 'DFA', 'spread1', 'spread2', 'D2', 'PPE']




*   Target Value :



 The target value can be said to be the dependent variable or the outcome that the model is trying to predict

 According to this model, I have to find the man who is a parkinson patient or not. If he is a parkinson patient = 1 and 0 otherwise.





*    Samples :



 Data included raws are the data samples.
 There are  195 samples(raws) in my dataset.
 Those are contain details of some people.



*   Performance of Algorithms



In KNN algorithm model gave 82.05% accuracy.

For decision tree and Support vector machine algorithm, my model gives 100% accuracy.